packages/compass/src/main/application.ts

```diff
@@ -186,7 +189,8 @@ class CompassApplication {
       let hasLegacyConnections: boolean;
       try {
         hasLegacyConnections =
-          (await ConnectionStorage.getLegacyConnections()).length > 0;
+          (await getCompassMainConnectionStorage().getLegacyConnections())
+            .length > 0;
```

Comment:
Not something to do an immediate action on, but relevant to something we discussed: seems like the getLegacyConnections belongs more in the "registry" hook, not in the storage, and the only reason we use it here is for telemetry. Maybe we can change the way telemetry reports it to remove the dependency (but would need to communicate this first to product)